File: src/algorithms/search/linear/js/linearSearch.js

/**
 * 
 * @param {Array} arr 
 * @param {Number} searchedNum 
 */
function linearSearch(arr, searchedNum){
    for(let i = 0; i < arr.length; i++){
        if(arr[i] === searchedNum){
            return i; 
        }
    }

    return -1; 
}


const arr = [20,10,4,10,42,0,21, -1]

console.log(linearSearch(arr, 3));

File: src/algorithms/search/linear/java/linearSearch.java
package algorithms.search.linear.java;

/** */
public class linearSearch {
    public static int run(int[] arr, int searchedNumber){

        for(int i = 0; i < arr.length; i++){
            // If the number at the specific index matches the searched number
            if(searchedNumber == arr[i] ){
                // Return the index of the number
                return i; 
            }
        }

        
        return -1;
    }; 
}